runtime/onert/core/src/ir/train/TrainableGraph.cc

```diff
@@ -125,9 +126,61 @@ const ITrainableOperation &TrainableGraph::operation(OperationIndex index) const
   return dynamic_cast<const ITrainableOperation &>(_graph.operations().at(index));
 }
 
+void TrainableGraph::assertValidTopologicalOrder(const std::vector<ir::OperationIndex> &order,
+                                                 const std::string &order_type = "forward") const
+{
+  auto compare = [](const ir::OperationIndex &i, const ir::OperationIndex &j) -> bool {
+    return i.value() < j.value();
+  };
+
+  std::map<ir::OperationIndex, uint32_t, decltype(compare)> position(compare);
```

Comment:
Q: Isn't it enough? 

```suggestion
  std::map<ir::OperationIndex, uint32_t, decltype(compare)> position;
```

Reply:
This syntax is valid since C++20 due to [default constructor for lambdas with no capture](https://en.cppreference.com/w/cpp/language/lambda#:~:text=If%20no%20captures%20are%20specified%2C%20the%20closure%20type%20has%20a%20defaulted%20default%20constructor.%20Otherwise%2C%20it%20has%20no%20default%20constructor%20(this%20includes%20the%20case%20when%20there%20is%20a%20capture%2Ddefault%2C%20even%20if%20it%20does%20not%20actually%20capture%20anything).). I believe it will CE currently due to `decltype(compare)` not having default constructor.